packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -477,6 +508,19 @@ export class ElectronMainApplication {
         app.on('will-quit', this.onWillQuit.bind(this));
         app.on('second-instance', this.onSecondInstance.bind(this));
         app.on('window-all-closed', this.onWindowAllClosed.bind(this));
+
+        ipcMain.on('titleBarStyle-changed', ({ sender }, titleBarStyle: string) => {
+            this.frame = titleBarStyle === 'native';
+            this.saveWindowState(BrowserWindow.fromId(sender.id));
+        });
+
+        ipcMain.on('restart', ({ sender }) => {
+            this.restart(sender.id);
+        });
+
+        ipcMain.on('request-titleBarStyle', ({ sender }) => {
+            sender.send('original-titleBarStyle', this.originalFrames.get(sender.id) ? 'native' : 'custom');
```

Comment:
consider extracting `request-titleBarStyle` and `original-titleBarStyle` (or `titleBarStyleAtStartup`) to constants to be used here and in the code that makes the request from the `electron-browser` code.